jwst/saturation/saturation.py

```diff
@@ -160,6 +165,29 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                scigp1 = x_irs2.from_irs2(data[ints, 0, :, :], irs2_mask, detector)
+                mask = scigp1 / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                # Identify groups with suspiciously large values in the second group
+                scigp2 = x_irs2.from_irs2(data[ints, 1, :, :], irs2_mask, detector)
+                mask &= scigp2 > sat_thresh / len(read_pattern[1])
+
+                # Identify groups that are saturated in the third group
+                gp3mask = np.where(flag_temp & SATURATED, True, False)
+                mask &= gp3mask
+
+                # now, flag any pixels that border saturated pixels
+                if n_pix_grow_sat > 0:
+                    mask = adjacency_sat(mask, SATURATED, n_pix_grow_sat)
```

Comment:
I don't think we can call it like this; 'mask' is an array of True/False whereas adjacency_sat expects input of actual dq flag values.  It should work though to say 
if n_pix_grow_sat > 0:
                    dq_temp = adjacency_sat(dq_temp, SATURATED, n_pix_grow_sat)
right after line 188 before the call to x_irs2.to_irs2